integration/spark/src/main/scala/org/apache/spark/sql/execution/command/management/CommonLoadUtils.scala

```diff
@@ -503,7 +503,20 @@ object CommonLoadUtils {
       if (numPartitions <= 0) {
         numPartitions = partitionsLen
       }
-      val sortColumns = attributes.take(table.getSortColumns.size())
+      val sortColumns =
+      if (table.isHivePartitionTable) {
+        // In case of partition column as sort column, attribute will not have it in the front.
+        // so need to look up the attribute and prepare
+        var sortColsAttr: Seq[AttributeReference] = Seq.empty
```

Comment:
use ArrayBuffer instead of Seq to make it mutable

Reply:
ok. done